### PR TITLE
Builder tweaks

### PR DESCRIPTION
**Pull request by kflynn**
- Update to Alpine 3.10 (and thus Python 3.7), and update to the Envoy from Ambassador 0.83.0.
- Add the `kubectl` binary
- Update to PyYAML 5.1, and build it in the `builder` container, so that we can use the C parser and dumper rather than the pure Python implementations.

### CODE REVIEW

File: builder/Dockerfile [hunk 1]
```diff
@@ -53,6 +53,18 @@ COPY --from=envoy /usr/local/bin/envoy /usr/local/bin/envoy
 
 ENV KUBECONFIG=/buildroot/kubeconfig.yaml
 
+# The YAML parser is... special. First, we need to install libyaml and Cython. No,
+# I don't know why it's called yaml-dev.
+RUN apk --no-cache add yaml-dev cython
```
Comment (LukeShu) [inline builder/Dockerfile]: Can this be part of the main `apk add`?

File: builder/Dockerfile [hunk 2]
```diff
@@ -53,6 +53,18 @@ COPY --from=envoy /usr/local/bin/envoy /usr/local/bin/envoy
...
+
+# After that, we can download !*@&#!*@&# PyYAML...
+RUN curl -O -L http://pyyaml.org/download/pyyaml/PyYAML-5.1.tar.gz
+RUN tar xzf PyYAML-5.1.tar.gz
```
Comment (LukeShu) [inline builder/Dockerfile]: I think the reason that CI is unhappy is that PyYAML is getting extracted over our sources in `/buildroot`.  Perhaps do this in `/tmp/pyyaml`?

File: builder/Dockerfile (hunk 1 above)
Comment (kflynn) [reply builder/Dockerfile]: Done.

File: builder/Dockerfile (hunk 2 above)
Comment (kflynn) [reply builder/Dockerfile]: Oh, nice catch! Done, and I folded the download/build/install `RUN`s together for you too...

**Comment by kflynn** [comment]
Another one I'm going to take even with the failing `watt` test.

**Pull request merged**